datamap/bloom/src/main/java/org/apache/carbondata/datamap/bloom/BloomCoarseGrainDataMapFactory.java

```diff
@@ -218,56 +218,46 @@ public DataMapBuilder createBuilder(Segment segment, String shardName,
         this.bloomFilterSize, this.bloomFilterFpp, bloomCompress);
   }
 
-  /**
-   * returns all shard directories of bloom index files for query
-   * if bloom index files are merged we should get only one shard path
-   */
-  private Set<String> getAllShardPaths(String tablePath, String segmentId) {
-    String dataMapStorePath = CarbonTablePath.getDataMapStorePath(
-        tablePath, segmentId, dataMapName);
-    CarbonFile[] carbonFiles = FileFactory.getCarbonFile(dataMapStorePath).listFiles();
-    Set<String> shardPaths = new HashSet<>();
+
+  private boolean isAllShardsMerged(String dmSegmentPath) {
+    boolean mergeShardExist = false;
     boolean mergeShardInprogress = false;
-    CarbonFile mergeShardFile = null;
+    CarbonFile[] carbonFiles = FileFactory.getCarbonFile(dmSegmentPath).listFiles();
     for (CarbonFile carbonFile : carbonFiles) {
-      if (carbonFile.getName().equals(BloomIndexFileStore.MERGE_BLOOM_INDEX_SHARD_NAME)) {
-        mergeShardFile = carbonFile;
-      } else if (carbonFile.getName().equals(BloomIndexFileStore.MERGE_INPROGRESS_FILE)) {
+      String fileName = carbonFile.getName();
+      if (fileName.equals(BloomIndexFileStore.MERGE_BLOOM_INDEX_SHARD_NAME)) {
+        mergeShardExist = true;
+      } else if (fileName.equals(BloomIndexFileStore.MERGE_INPROGRESS_FILE)) {
         mergeShardInprogress = true;
```

Comment:
If MERGE_INPROGRESS_FILE exists, shard's index file will be deleted  sometime, so this scene need to be focus on, but this question shows up before this PR

Reply:
Yes, you are right. We need to fix this. If we allow to use bloom filter when the index files are merging, maybe any IO Exception will occur in following steps when the merging is done. 

Some simple ideas for this:
1. datamap do not choose bloom when merging is under action
2. change the pruning logic to segment independent, any datamap excepts default datamap can reject or fail the segment pruning ( by return null or ?), and no more result blocklet intersection for this datamap, such that this does not affect final result

Reply:
One more idea is that we can delay the deletion of original shards in query, referring to segment management. That is when mergeShard exists and no merge inprogress file in a query, we can assure to delete original shards safely.